src/mapgen.cpp

```diff
@@ -754,7 +754,7 @@ class jmapgen_vehicle : public jmapgen_piece {
         if( !x_in_y( chance.get(), 100 ) ) {
             return;
         }
-        m.add_vehicle( type, x, y, rotation[rng(0, rotation.size()-1)], fuel, status );
+        m.add_vehicle( type, point(x, y), rotation[rng(0, rotation.size()-1)], fuel, status );
```

Comment:
if there is a 3d version of the second argument, use that instead (unless it takes more code, like if the function only takes [x, y] as arguments to begin with).